apps/arm/src/main.cpp

```diff
@@ -167,14 +170,72 @@ static mbed_error_status_t setToolTipDeployment(CANMsg &msg) {
   }
 }
 
-// Enable or disable PID tuning mode10
+// Enable or disable PID tuning mode
+// Manually send CAN message to SET_PID_TUNING_MODE with appropriate boolean
 static mbed_error_status_t setPIDTuningMode(CANMsg &msg) {
-  return MBED_SUCCESS;  // TODO
+  msg.getPayload(allowPIDParamTuning);
+  return MBED_SUCCESS;
 }
 
 // Configure PID parameters
 static mbed_error_status_t setPIDParameter(CANMsg &msg) {
-  return MBED_SUCCESS;  // TODO
+  printf("Received request to update PID params over CAN\n");
+  if (!allowPIDParamTuning) {
+    printf("Unable to update PID params over CAN. Ensure arm is in a safe state and update Tuning Mode.\n");
+    printf("To allow PID param tuning over CAN, send true to CAN address SET_PID_TUNING_MODE\n");
+    return MBED_ERROR_ASSERTION_FAILED;
+  }
+  /** ------------------------------------
+   * | 32 bits  | 8 bits   |  8 bits     |
+   * | data     | vel/pos  |  Actuator ID|
+   *  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^*/
+  HWBRIDGE::ARM::PID::tuningApiPayload payload;
+  msg.getPayload(payload);
+  printf("payload: %s", HWBRIDGE::ARM::PID::str(payload).c_str());
+  ActuatorController *temp = nullptr;
+  switch (payload.actuatorID) {
+    case HWBRIDGE::ARM::ACTUATOR::TURNTABLE:
+      temp = &turnTableActuator;
```

Comment:
Almost always prefer to use references over pointers in C++.

Reply:
i dont think & is right here bc we dont initalize right away

Reply:
right, use smart pointers then. probs unique ptr